wled00/FX.cpp

```diff
@@ -7110,6 +7110,42 @@ uint16_t mode_waterfall(void) {                   // Waterfall. By: Andrew Tulin
 static const char _data_FX_MODE_WATERFALL[] PROGMEM = "Waterfall@!,Adjust color,Select bin,Volume (min);!,!;!;1f;c2=0,m12=2,si=0"; // Circles, Beatsin
 
 
+/*
+ * Uses the average value from user-selected FFT buckets to select a single
+ * color from a palette.
+ */
+uint16_t mode_freqpalette() {
+  um_data_t *um_data;
+  if (!usermods.getUMData(&um_data, USERMOD_ID_AUDIOREACTIVE)) {
+    // add support for no audio
+    um_data = simulateSound(SEGMENT.soundSim);
+  }
+
+  // Get minimum and maximum FFT bands to consider
+  uint8_t minBand = constrain(SEGMENT.custom1 / 16, 0, 15);
+  uint8_t maxBand = constrain(SEGMENT.custom2 / 16, 0, 15);
+  if (maxBand < minBand) {
+    uint8_t temp = minBand;
+    minBand = maxBand;
+    maxBand = temp;
+  }
+
+  // Average the bands' values, scaled by user-specified sensitivity
+  uint32_t value = 0;
+  uint8_t *fftResult = (uint8_t*)um_data->u_data[2];
+  for (uint8_t band = minBand; band <= maxBand; ++band) {
+    value += fftResult[band] * SEGMENT.intensity;
+  }
+  value /= (maxBand - minBand + 1) * (256 / 8);
+  value = constrain(value, 0, 255);
+
+  SEGMENT.fill(SEGMENT.color_from_palette(value, false, false, 0));
```

Comment:
Would be nice to have two options here (based on checkbox)
A) fill with a single color
B) fill with palette color up to the value of volume (similar to the gravCentric effect), like a VU meter.

I think this will add even more "creative potential" to your effect. You can implement case B just as left-to-right VU meter. Centering, reverse direction, and mirroring can be done with segment options so no need to care about that in the effect itself..